File: back/src/main/java/com/funstat/domain/InstrId.java
package com.funstat.domain;

import io.swagger.annotations.ApiModelProperty;

public class InstrId {
    @ApiModelProperty(required = true)
    public String id;
    @ApiModelProperty(required = true)
    public String name;
    @ApiModelProperty(required = true)
    public String market;
    @ApiModelProperty(required = true)
    public String code;
    @ApiModelProperty(required = true)
    public String source;

    public InstrId(String id, String name, String market, String code, String source) {
        this.id = id;
        this.name = name;
        this.market = market;
        this.code = code;
        this.source = source;
    }

    @ApiModelProperty(required = true)
    public String getId() {
        return id;
    }

    @ApiModelProperty(required = true)
    public String getName() {
        return name;
    }

    @ApiModelProperty(required = true)
    public String getMarket() {
        return market;
    }

    @ApiModelProperty(required = true)
    public String getCode() {
        return code;
    }

    @ApiModelProperty(required = true)
    public String getSource() {
        return source;
    }

    public InstrId() {
        this(null,null,null,null, null);
    }

    @Override
    public String toString() {
        return "InstrId{" +
                "id='" + id + '\'' +
                ", name='" + name + '\'' +
                ", market='" + market + '\'' +
                ", code='" + code + '\'' +
                '}';
    }
}

File: back/src/main/java/com/funstat/domain/InstrId.kt
package com.funstat.domain

import java.math.BigDecimal


data class InstrId(val id: String = "N/A",
                   val name: String = "N/A",
                   val market: String = "N/A",
                   val code: String = "N/A",
                   val source: String = "N/A",
                   val minPriceIncr: BigDecimal = BigDecimal(0.1),
                   val lot: Int = 1


){
    companion object{
        fun dummyInstrument(ticker : String) : InstrId{
            return InstrId(ticker,ticker,"",ticker,"dummy")
        }
    }
}


